pkg/resolution/resolver/framework/interface.go

```diff
@@ -43,13 +43,25 @@ type Resolver interface {
 	// request and should return an error if any are missing or invalid.
 	ValidateParams(ctx context.Context, params []pipelinev1.Param) error
 
+	// ValidateUrl is given the url from a resource
+	// request and should return an error if the resolver cannot resolve it.
+	ValidateUrl(ctx context.Context, url string) error
+
 	// Resolve receives the parameters passed via a resource request
 	// and returns the resolved data along with any annotations
 	// to include in the response. If resolution fails then an error
 	// should be returned instead. If a resolution.Error
 	// is returned then its Reason and Message are used as part of the
 	// response to the request.
 	Resolve(ctx context.Context, params []pipelinev1.Param) (ResolvedResource, error)
+
+	// ResolveUrl receives the url passed via a resource request
+	// and returns the resolved data along with any annotations
+	// to include in the response. If resolution fails then an error
+	// should be returned instead. If a resolution.Error
+	// is returned then its Reason and Message are used as part of the
+	// response to the request.
+	ResolveUrl(ctx context.Context, url string) (ResolvedResource, error)
```

Comment:
Would it make more sense to have a func that takes in a ResolutionRequestSpec so we don't have to make different Resolve functions?

e.g.

```go
ValidateRequest(ctx context.Context, req *v1beta1.ResolutionRequestSpec) error
ResolveRequest(ctx context.Context, req *v1beta1.ResolutionRequestSpec) (ResolvedResource, error)
```

This would make it easier to add additional fields in a backwards compatible way later.

Reply:
Sure, I think we can do that. But we will have to leave the existing functions as is so that we don't make backwards incompatible changes.

Reply:
Fixed. Need to improve coverage.

Reply:
Fixed!